Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1478,10 +1478,13 @@ internal extension Purchases {
 
 private extension Purchases {
 
-    func handleCustomerInfoChanged(_ customerInfo: CustomerInfo) {
-        self.trialOrIntroPriceEligibilityChecker.clearCache()
-        self.purchasedProductsFetcher?.clearCache()
-        self.delegate?.purchases?(self, receivedUpdated: customerInfo)
+    func handleCustomerInfoChanged(from old: CustomerInfo?, to new: CustomerInfo) {
+        if old != nil {
```

Comment:
This is the core of this change. We no longer clear the caches for the first time we receive a customer info change.